CMakeLists.txt

```diff
@@ -56,6 +56,12 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
         )
     endif()
     add_compile_options(-Wno-shift-negative-value)
+    if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24.0")
+      option(CMAKE_COMPILE_WARNING_AS_ERROR "Treat compiler warnings as errors." ON)
+      mark_as_advanced(CMAKE_COMPILE_WARNING_AS_ERROR)
```

Comment:
We don't need this extra option, if `--compile-no-warning-as-error` has the same effect.

Reply:
The problem is that `--compile-no-warning-as-error` overrides the `COMPILE_WARNING_AS_ERROR` target property, but you still need to enable the `COMPILE_WARNING_AS_ERROR` if you actually want to pass `-Werror` to the compilation unless `--compile-no-warning-as-error` is passed.

Setting `CMAKE_COMPILE_WARNING_AS_ERROR` by default to `ON` is a way of doing that, an alternative is setting it as a normal variable instead of an option, or another is to set the property manually, like `set_property(TARGET ${PROJECT_NAME} PROPERTY COMPILE_WARNING_AS_ERROR ON)`.

Reply:
Ah, so you suggest to set it as a normal variable, and not as an option? Yes that would work, I will modify the PR tomorrow.